dds_web/web/user.py

```diff
@@ -193,12 +252,22 @@ def login():
                 flask.url_for("auth_blueprint.login", next=next)
             )  # Try login again
 
-        # Correct username and password --> log user in
-        flask_login.login_user(user)
-        flask.flash("Logged in successfully.")
+        # Correct credentials still needs 2fa
 
-        # Go to home page
-        return flask.redirect(next or flask.url_for("auth_blueprint.index"))
+        # Send 2fa token to user's email
+        hotp_value = user.generate_HOTP_token()
+        msg = dds_web.utils.create_one_time_password_email(user, hotp_value)
+        mail.send(msg)
+        flask.flash("One-Time Code has been sent to your primary email.")
```

Comment:
Instead of lines 258-261, use 

```
if dds_web.dds_web.security.auth.send_hotp_email(user):
    flask.flash("One-Time Code has been sent to your primary email.")
```

In order for this to work, update in auth.py

```
def send_hotp_email(user):
    if not user.hotp_issue_time or (
        user.hotp_issue_time
        and (dds_web.utils.current_time() - user.hotp_issue_time > datetime.timedelta(minutes=15))
    ):
        hotp_value = user.generate_HOTP_token()
        msg = dds_web.utils.create_one_time_password_email(user, hotp_value)
        mail.send(msg)
        return True
    return False
```

Reply:
Yes I think this is a good addition!